File: domain/src/main/java/com/ortin/gesturetranslator/domain/usecases/RecognizeCoordinateUseCase.java
package com.ortin.gesturetranslator.domain.usecases;

import com.ortin.gesturetranslator.domain.models.CoordinateClassification;
import com.ortin.gesturetranslator.domain.models.ImageDetected;
import com.ortin.gesturetranslator.domain.repository.RecognizeCoordinateRepository;

public class RecognizeCoordinateUseCase {
    private final RecognizeCoordinateRepository recognizeCoordinateRepository;

    public RecognizeCoordinateUseCase(RecognizeCoordinateRepository recognizeCoordinateRepository) {
        this.recognizeCoordinateRepository = recognizeCoordinateRepository;
    }

    public CoordinateClassification execute(ImageDetected imageDetected) {
        return recognizeCoordinateRepository.recognise(imageDetected);
    }
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/usecases/RecognizeCoordinateUseCase.kt
package com.ortin.gesturetranslator.domain.usecases

import com.ortin.gesturetranslator.domain.models.CoordinateClassification
import com.ortin.gesturetranslator.domain.models.ImageDetected
import com.ortin.gesturetranslator.domain.repository.RecognizeCoordinateRepository
import javax.inject.Inject
import javax.inject.Singleton

interface RecognizeCoordinateUseCase : UseCase<ImageDetected, CoordinateClassification>

@Singleton
class RecognizeCoordinateUseCaseImpl @Inject constructor(
    private val recognizeCoordinateRepository: RecognizeCoordinateRepository
) : RecognizeCoordinateUseCase {
    override fun invoke(data: ImageDetected): CoordinateClassification =
        recognizeCoordinateRepository.recognise(data)
}
